test_truststore.py

```diff
@@ -21,19 +22,74 @@
 
 successful_hosts = pytest.mark.parametrize("host", ["example.com", "1.1.1.1"])
 
+
+@dataclass
+class FailureHost:
+    host: str
+    error_messages: list[str]
+
+
 failure_hosts_list = [
-    "wrong.host.badssl.com",
-    "expired.badssl.com",
-    "self-signed.badssl.com",
-    "untrusted-root.badssl.com",
-    "superfish.badssl.com",
+    FailureHost(
+        host="wrong.host.badssl.com",
+        error_messages=[
+            # OpenSSL
+            "Hostname mismatch, certificate is not valid for 'wrong.host.badssl.com'",
+            # macOS
+            "certificate name does not match",
+        ],
+    ),
+    FailureHost(
+        host="expired.badssl.com",
+        error_messages=[
+            # OpenSSL
+            "certificate has expired",
+            # macOS
+            "“*.badssl.com” certificate is expired",
+        ],
+    ),
+    FailureHost(
+        host="self-signed.badssl.com",
+        error_messages=[
+            # OpenSSL
+            "self signed certificate",
+            # macOS
+            "“*.badssl.com” certificate is not trusted",
+        ],
+    ),
+    FailureHost(
+        host="untrusted-root.badssl.com",
+        error_messages=[
+            # OpenSSL
+            "self signed certificate in certificate chain",
+            # macOS
+            "“BadSSL Untrusted Root Certificate Authority” certificate is not trusted",
+        ],
+    ),
+    FailureHost(
+        host="superfish.badssl.com",
+        error_messages=[
+            # OpenSSL
+            "unable to get local issuer certificate",
+            # macOS
+            "“superfish.badssl.com” certificate is not trusted",
+        ],
+    ),
 ]
 
 if platform.system() != "Linux":
-    failure_hosts_list.append("revoked.badssl.com")
+    failure_hosts_list.append(
+        FailureHost(
+            host="revoked.badssl.com",
+            error_messages=[
+                # macOS
+                "“revoked.badssl.com” certificate is revoked"
+            ],
+        )
+    )
 
 failure_hosts = pytest.mark.parametrize(
-    "host",
+    "failure",
     failure_hosts_list,
 )
```

Comment:
You could set `ids=operator.attrgetter("host")` here so we still get the host identified in the test case names instead of failure0, failure1, etc.

https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids

Reply:
This is superb!! I'll be using this in the future too.